src/BloomFilter.cpp

```diff
@@ -119,89 +132,25 @@ static unsigned int doubleHash(unsigned int hash1, unsigned int hash2, unsigned
     }
 }
 
-void BloomFilter::writeToFile(string path) {
+void BloomFilter::writeToFile(const string &path) {
     basic_ofstream<BlockType> out(path.c_str(), ofstream::binary);
-    writeVectorToStream(bloomVector, out);
+    writeToStream(out);
 }
 
 void BloomFilter::writeToStream(BinaryOutputStream &out) {
-    writeVectorToStream(bloomVector, out);
+    out.write(&bloomVector[0], bloomVector.size() * sizeof(BlockType));
 }
 
-static void writeVectorToStream(vector<bool> &bloomVector, BinaryOutputStream &out) {
-
-    const size_t elements = bloomVector.size();
-    out.put(((elements & 0x000000ff) >> 0));
-    out.put(((elements & 0x0000ff00) >> 8));
-    out.put(((elements & 0x00ff0000) >> 16));
-    out.put(((elements & 0xff000000) >> 24));
-
-    const size_t bitsPerBlock = sizeof(BlockType) * 8;
-    for (size_t i = 0; i < elements / bitsPerBlock; i++) {
-        const BlockType buffer = pack(bloomVector, i, bitsPerBlock);
-        out.put(buffer);
-    }
-
-    const size_t bitsInLastBlock = elements % bitsPerBlock;
-    if (bitsInLastBlock > 0) {
-        const size_t lastBlock = elements / bitsPerBlock;
-        const BlockType buffer = pack(bloomVector, lastBlock, bitsInLastBlock);
-        out.put(buffer);
-    }
-}
-
-static BlockType pack(const vector<bool> &filter, size_t block, size_t bits) {
-
-    const size_t sizeOfTInBits = sizeof(BlockType) * 8;
-    assert(bits <= sizeOfTInBits);
-    BlockType buffer = 0;
-    for (size_t j = 0; j < bits; ++j) {
-        const size_t offset = (block * sizeOfTInBits) + j;
-        const BlockType bit = filter[offset] << j;
-        buffer |= bit;
-    }
-    return buffer;
-}
-
-static vector<bool> readVectorFromFile(const string &path) {
+static vector<BlockType> readVectorFromFile(const string &path) {
     basic_ifstream<BlockType> inFile(path, ifstream::binary);
     return readVectorFromStream(inFile);
 }
 
-static vector<bool> readVectorFromStream(BinaryInputStream &in) {
-
-    const size_t component1 = in.get() << 0;
-    const size_t component2 = in.get() << 8;
-    const size_t component3 = in.get() << 16;
-    const size_t component4 = in.get() << 24;
-    const size_t elementCount = component1 + component2 + component3 + component4;
-
-    vector<bool> bloomVector(elementCount);
-    const size_t bitsPerBlock = sizeof(BlockType) * 8;
-    const size_t fullBlocks = elementCount / bitsPerBlock;
-    for (size_t i = 0; i < fullBlocks; ++i) {
-        const size_t offset = i * bitsPerBlock;
-        unpackIntoVector(bloomVector, offset, bitsPerBlock, in);
-    }
-
-    const size_t bitsInLastBlock = elementCount % bitsPerBlock;
-    if (bitsInLastBlock > 0) {
-        const size_t offset = bitsPerBlock * fullBlocks;
-        unpackIntoVector(bloomVector, offset, bitsInLastBlock, in);
-    }
-
+static vector<BlockType> readVectorFromStream(BinaryInputStream &in) {
+    vector<BlockType> bloomVector((istreambuf_iterator<BlockType>(in)), istreambuf_iterator<BlockType>());
```

Comment:
Read times are much faster than before (around 35%) but if you can think of any tweaks to make this read even faster let me know!